byron/ledger/executable-spec/test/Ledger/Delegation/Properties.hs

```diff
@@ -325,88 +331,50 @@ dblockTracesAreClassified = withTests 200 $ property $ do
 
 -- | Extract the delegation certificates in the blocks, in the order they would
 -- have been applied.
+traceDCertsByBlock :: Trace DBLOCK -> [[DCert]]
+traceDCertsByBlock tr = _blockCerts <$> traceSignals OldestFirst tr
+
+-- | Flattended list of DCerts for the given Trace
 traceDCerts :: Trace DBLOCK -> [DCert]
-traceDCerts tr = concat $ _blockCerts <$> traceSignals OldestFirst tr
+traceDCerts = concat . traceDCertsByBlock
 
 relevantCasesAreCovered :: Property
 relevantCasesAreCovered = withTests 400 $ property $ do
   let tl = 1000
   tr <- forAll (trace @DBLOCK tl)
 
-  -- 70% of the traces must contain are as many delegation certificates as
-  -- blocks.
-  cover 50
+  -- 40% of the traces must contain as many delegation certificates as blocks.
+  cover 40
```

Comment:
Can you explain why this change? Were generators modified that begged for updating these thresholds?

Reply:
The threshold lied very close to the values that we got in the runs. Like you noticed, these thresholds are arbitrary, so these kind of adaptations would be needed. But indeed, we should ask for explanations whenever we change these :)

Reply:
these settings are still a bit brittle since we're still changing Chain trace generation, it will stabilise in due